src/main/java/com/store/bookstore/validation/match/FieldMatchValidator.java

```diff
@@ -0,0 +1,62 @@
+package com.store.bookstore.validation.match;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+
+    public static final SpelExpressionParser EXPRESSION_PARSER = new SpelExpressionParser();
+    private String[] fieldNames;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.fieldNames = constraintAnnotation.value();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext constraintValidatorContext) {
```

Comment:
it could be simplier
```suggestion
    @Override
    public boolean isValid(Object value, ConstraintValidatorContext constraintValidatorContext) {
        Object field = new BeanWrapperImpl(value).getPropertyValue(this.field);
        Object fieldMatch = new BeanWrapperImpl(value).getPropertyValue(this.fieldMatch);
        return Objects.equals(field, fieldMatch);
    }
```